mirage/seed_image/catalog_seed_image.py

```diff
@@ -3179,6 +3189,12 @@ def checkParams(self):
 
                 self.params['simSignals']['psfpath'] = os.path.join(self.params['simSignals']['psfpath'], pathaddition)
                 self.psfname = os.path.join(self.params['simSignals']['psfpath'], psfname)
+                # In the NIRISS AMI mode case, replace NIS by NIS_NRM as the PSF 
+                # files are in a separate directory and have the altered file names
+                # compared to imaging.  Hence one can point to the same base 
+                # directory to run both imaging and NRM models.
+                if self.params['Inst']['mode'] in ['ami']:
+                    self.psfname = self.psfname.replace('NIS','NIS_NRM')
```

Comment:
I'm a little concerned about getting the PSF filename this way. This assumes that the NRM files are in a particular location relative to the imaging files. What if someone creates a new or specialized NRM library of some sort that they want to test? (I'm not sure how realistic that situation would be.) 

But, I can see how really this is probably the best solution given the assumptions about filenames and directory structure that mirage makes when constructing the PSF filename. 

My hope is that when the ePSF changes are introduced, that the PSF library input value will be the full path to the directory containing the files (no assumption of subdirectory structure), since there should be many fewer and more easily organized files in that case. So for example, the yaml inputs for imaging and NRM in that case would be something like:
/myfiles/data/psf_libraries/NIRISS/imaging/  and 
/myfiles/data/psf_libraries/NIRISS/NRM/
With this method, there's no built-in assumption of where files are located nor what kind of subdirectories are under the given path.

Reply:
I did not think about it in these terms, but of course you are right that this requires the specific directory structure for the PSF files.  I think we just need to document it in the code, and assume that when we move to ePSF that this will no longer be an issue.  One already requires the specific directory structure for the PSF libraries, and so extending this one level for imaging versus NRM PSF libraries is not a big change in "philosophy".  And it simplifies using the yaml files as one only needs to change the mode from imaging to AMI and everything works.  Hence I will add some comments to the code but not change the mechanics.